StreamReader.cpp

```diff
@@ -37,34 +37,58 @@ void FileStreamReader::Seek(uint64_t position) {
 	file.seekg(position, std::ios_base::beg);
 }
 
-void FileStreamReader::SeekRelative(int offset) {
+void FileStreamReader::SeekRelative(int64_t offset) {
 	file.seekg(offset, std::ios_base::cur);
 }
 
 
-MemoryStreamReader::MemoryStreamReader(char* buffer, size_t size) {
+MemoryStreamReader::MemoryStreamReader(char* buffer, uint64_t size) {
 	streamBuffer = buffer;
 	streamSize = size;
-	offset = 0;
+	position = 0;
 }
 
-void MemoryStreamReader::Read(char* buffer, size_t size) 
+void MemoryStreamReader::Read(char* buffer, uint64_t size) 
 {
-	if (offset + size > streamSize) {
+	if (position + size > streamSize) {
 		throw std::runtime_error("Size of bytes to read exceeds remaining size of buffer.");
 	}
 
-	memcpy(buffer, streamBuffer + offset, size);
-	offset += size;
+	// std::memcpy takes a _Size to copy of size_t. 
+	// Depending on the system, size_t may not be able to store the max value of uint64_t. 
+	// SIZE_MAX is the largest value that may be stored in size_t for the given system.
+	do {
+		size_t nextReadLength = SIZE_MAX;
+		// Check if remaining value of size will fit in a size_t.
+		if (size < SIZE_MAX) {
+			nextReadLength = static_cast<size_t>(size);
+		}
+
+		std::memcpy(buffer, streamBuffer + position, nextReadLength);
+		position += nextReadLength;
+		size -= nextReadLength;
+	} while (size != 0); // Continue reading in chunks until all data has been copied.
```

Comment:
I think we may be better off with the `memcpy`. If the range different between `size_t` and `uint64_t` is an issue, perhaps we can raise an exception in that case.